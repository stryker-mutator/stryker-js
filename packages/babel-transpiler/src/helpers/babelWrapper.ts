/**
 * This wrapper is needed because babel/core only exports
 * non-stubbable es6 properties. See node_modules/@babel/core/lib/index.js
 */
export * from '@babel/core';
