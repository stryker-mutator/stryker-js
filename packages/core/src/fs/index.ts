export * from './file-system.js';
export * from './project.js';
export * from './project-file.js';
export * from './project-reader.js';
