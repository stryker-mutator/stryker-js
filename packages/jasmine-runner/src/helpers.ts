import { TestResult, TestStatus } from '@stryker-mutator/api/test-runner';
import jasmine from 'jasmine';

export const helpers = {
  toStrykerTestResult(specResult: jasmine.SpecResult, timeSpentMs: number): TestResult {
    const baseResult = {
      id: specResult.id.toString(),
      name: specResult.fullName,
      timeSpentMs,
    };
    if (specResult.status === 'disabled' || specResult.status === 'pending' || specResult.status === 'excluded') {
      return {
        ...baseResult,
        status: TestStatus.Skipped,
      };
    } else if (!specResult.failedExpectations || specResult.failedExpectations.length === 0) {
      return {
        ...baseResult,
        status: TestStatus.Success,
      };
    } else {
      return {
        ...baseResult,
        status: TestStatus.Failed,
        failureMessage: specResult.failedExpectations.map((failedExpectation) => failedExpectation.message).join(','),
      };
    }
  },
  createJasmine(options: jasmine.JasmineOptions): jasmine {
    return new jasmine(options);
  },
};
