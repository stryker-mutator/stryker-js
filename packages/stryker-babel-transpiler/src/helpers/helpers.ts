import * as path from 'path';

export function toJSFileName(fileName: string) {
  const fileNameInfo = path.parse(fileName);
  const newFileName = path.join(fileNameInfo.dir, fileNameInfo.name + '.js');
  return newFileName;
}
