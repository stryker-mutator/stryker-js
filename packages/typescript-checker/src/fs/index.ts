export * from './hybrid-file-system';
