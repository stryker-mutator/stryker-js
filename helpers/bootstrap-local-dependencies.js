// @ts-check
import { LocalInstaller, progress } from 'install-local';
import fs from 'fs';
import glob from 'glob';
import path from 'path';

console.log('starting installation of local dependencies');

/**
 * @param {string} pattern 
 * @param {glob.IOptions} options 
 * @returns {Promise<string[]>}
 */
function globAsPromised(pattern, options) {
  return new Promise((res, rej) => {
    glob(pattern, options, (err, matches) => {
      if (err) {
        rej(err);
      } else {
        res(matches);
      }
    });
  });
}

/**
 * @typedef Package
 * @property {{ [name: string]: string }} localDependencies

/**
 * Installs local dependencies in one go,
 * reads the package.json files and installs the local dependencies marked there.
 * The packageJsonGlobs default to ./package.json and test/* /package.json files 
 * @param {string} directory
 * @param {string[]} packageJsonGlobs
 * @returns {Promise<void>}
 */
export async function bootstrapLocalDependencies(directory, packageJsonGlobs = ['package.json', 'test/*/package.json']) {
  console.log(`bootstrap ${path.resolve(directory)} (using ${packageJsonGlobs.join(',')})`);
  const files = (await Promise.all(packageJsonGlobs.map((globPattern) => globAsPromised(globPattern, { cwd: path.resolve(directory) })))).flat();
  const packages = await Promise.all(files.map(fileName => fs.promises.readFile(fileName, 'utf-8')
    .then((rawContent) => {
      /**
       * @type {Package}
       */
      const content = JSON.parse(rawContent);
      return { dir: path.dirname(fileName), content };
    })));
  /**
   * @type {import('install-local').ListByPackage}
   */
  const sourcesByTarget = {};
  for (const pkg of packages) {
    const localDeps = pkg.content.localDependencies;
    if (localDeps) {
      sourcesByTarget[path.resolve(pkg.dir)] = Object.keys(localDeps).map(key => path.resolve(pkg.dir, localDeps[key]));
    }
  }
  const localInstaller = new LocalInstaller(sourcesByTarget);
  progress(localInstaller);
  await localInstaller.install();
}
