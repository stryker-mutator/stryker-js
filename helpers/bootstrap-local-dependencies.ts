import { ListByPackage, LocalInstaller, progress } from 'install-local';
import * as fs from 'fs';
import glob = require('glob');
import path = require('path');

console.log('starting installation of local dependencies');

function globAsPromised(pattern: string, options: glob.IOptions) {
  return new Promise<string[]>((res, rej) => {
    glob(pattern, options, (err, matches) => {
      if (err) {
        rej(err);
      } else {
        res(matches);
      }
    });
  });
}

function readFile(fileName: string) {
  return new Promise<string>((res, rej) => fs.readFile(fileName, 'utf8', (err, content) => {
    if (err) {
      rej(err);
    } else {
      res(content);
    }
  }));
}

interface Package {
  localDependencies?: { [name: string]: string };
}

/**
 * Installs local dependencies in one go,
 * reads package.json and test/* /package.json files and installs the local dependencies marked there
 * @param directory the directory where the tests live
 */
export async function bootstrapLocalDependencies(directory: string) {
  console.log('bootstrap ' + path.resolve(directory));
  const files = await globAsPromised('{package.json,test/*/package.json}', { cwd: path.resolve(directory) });
  const packages = await Promise.all(files.map(fileName => readFile(fileName)
    .then(content => ({ dir: path.dirname(fileName), content: JSON.parse(content) as Package }))));
  const sourcesByTarget: ListByPackage = {};
  for (const pkg of packages) {
    const localDeps = pkg.content.localDependencies;
    if (localDeps) {
      sourcesByTarget[path.resolve(pkg.dir)] = Object.keys(localDeps).map(key => path.resolve(pkg.dir, localDeps[key]));
    }
  }
  const localInstaller = new LocalInstaller(sourcesByTarget);
  progress(localInstaller);
  await localInstaller.install();
}
