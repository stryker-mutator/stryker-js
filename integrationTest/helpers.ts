import * as path from 'path';
import * as fs from 'mz/fs';
import { expect } from 'chai';
import { ScoreResult } from 'stryker-api/report';

export async function readScoreResult(eventResultDirectory = path.resolve('reports', 'mutation', 'events')) {
    const allReportFiles = await fs.readdir(eventResultDirectory);
    const scoreResultReportFile = allReportFiles.find(file => !!file.match(/.*onScoreCalculated.*/));
    expect(scoreResultReportFile).ok;
    const scoreResultContent = await fs.readFile(path.resolve(eventResultDirectory, scoreResultReportFile || ''), 'utf8');
    return JSON.parse(scoreResultContent) as ScoreResult;
}