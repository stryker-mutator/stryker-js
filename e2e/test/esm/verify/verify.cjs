// @ts-check
const fs = require('fs');
const { expectMetricsJsonToMatchSnapshot, execStryker } = require('../../../helpers');

describe('esm', () => {
  beforeEach(async () => {
    await fs.promises.rm('reports', { recursive: true, force: true });
  });

  it('should be supported in the mocha runner', async () => {
    execStryker('stryker run --testRunner mocha');
    await assertStrykerRanCorrectly();
  });

  it('should be supported in the jasmine runner', async () => {
    execStryker('stryker run --testRunner jasmine');
    await assertStrykerRanCorrectly();
  });
});

async function assertStrykerRanCorrectly() {
  await expectMetricsJsonToMatchSnapshot();
}
