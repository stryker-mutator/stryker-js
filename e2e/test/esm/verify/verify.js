import fs from 'fs';

import { expectMetricsJsonToMatchSnapshot, execStryker } from '../../../helpers.js';

describe('esm', () => {
  beforeEach(async () => {
    await fs.promises.rm('reports', { recursive: true, force: true });
  });

  it('should be supported in the mocha runner', async () => {
    execStryker('stryker run --testRunner mocha');
    await assertStrykerRanCorrectly();
  });

  it('should be supported in the jasmine runner', async () => {
    execStryker('stryker run --testRunner jasmine');
    await assertStrykerRanCorrectly();
  });

  it('should be supported in the jest runner', async () => {
    execStryker('stryker run --testRunner jest --testRunnerNodeArgs "--experimental-vm-modules"');
    await assertStrykerRanCorrectly();
  });

  it('should be supported in the karma runner', async () => {
    execStryker('stryker run --testRunner karma');
    await assertStrykerRanCorrectly();
  });

  it('should be supported in the cucumber runner', async () => {
    execStryker('stryker run --testRunner cucumber');
    await assertStrykerRanCorrectly();
  });
});

async function assertStrykerRanCorrectly() {
  await expectMetricsJsonToMatchSnapshot();
}
