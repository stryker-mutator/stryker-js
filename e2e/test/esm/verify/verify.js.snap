// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`esm should be supported in the cucumber runner 1`] = `
Object {
  "compileErrors": 0,
  "ignored": 0,
  "killed": 5,
  "mutationScore": 55.55555555555556,
  "mutationScoreBasedOnCoveredCode": 71.42857142857143,
  "noCoverage": 2,
  "runtimeErrors": 0,
  "survived": 2,
  "timeout": 0,
  "totalCovered": 7,
  "totalDetected": 5,
  "totalInvalid": 0,
  "totalMutants": 9,
  "totalUndetected": 4,
  "totalValid": 9,
}
`;

exports[`esm should be supported in the jasmine runner 1`] = `
Object {
  "compileErrors": 0,
  "ignored": 0,
  "killed": 5,
  "mutationScore": 55.55555555555556,
  "mutationScoreBasedOnCoveredCode": 71.42857142857143,
  "noCoverage": 2,
  "runtimeErrors": 0,
  "survived": 2,
  "timeout": 0,
  "totalCovered": 7,
  "totalDetected": 5,
  "totalInvalid": 0,
  "totalMutants": 9,
  "totalUndetected": 4,
  "totalValid": 9,
}
`;

exports[`esm should be supported in the jest runner 1`] = `
Object {
  "compileErrors": 0,
  "ignored": 0,
  "killed": 5,
  "mutationScore": 55.55555555555556,
  "mutationScoreBasedOnCoveredCode": 71.42857142857143,
  "noCoverage": 2,
  "runtimeErrors": 0,
  "survived": 2,
  "timeout": 0,
  "totalCovered": 7,
  "totalDetected": 5,
  "totalInvalid": 0,
  "totalMutants": 9,
  "totalUndetected": 4,
  "totalValid": 9,
}
`;

exports[`esm should be supported in the karma runner 1`] = `
Object {
  "compileErrors": 0,
  "ignored": 0,
  "killed": 5,
  "mutationScore": 55.55555555555556,
  "mutationScoreBasedOnCoveredCode": 71.42857142857143,
  "noCoverage": 2,
  "runtimeErrors": 0,
  "survived": 2,
  "timeout": 0,
  "totalCovered": 7,
  "totalDetected": 5,
  "totalInvalid": 0,
  "totalMutants": 9,
  "totalUndetected": 4,
  "totalValid": 9,
}
`;

exports[`esm should be supported in the mocha runner 1`] = `
Object {
  "compileErrors": 0,
  "ignored": 0,
  "killed": 5,
  "mutationScore": 55.55555555555556,
  "mutationScoreBasedOnCoveredCode": 71.42857142857143,
  "noCoverage": 2,
  "runtimeErrors": 0,
  "survived": 2,
  "timeout": 0,
  "totalCovered": 7,
  "totalDetected": 5,
  "totalInvalid": 0,
  "totalMutants": 9,
  "totalUndetected": 4,
  "totalValid": 9,
}
`;
